### PR TITLE
src/mon/PGMap: raise POOL_APP_NOT_ENABLED after some time for new pool

### PR DESCRIPTION
**Pull request by guojidan**
Fixes: https://tracker.ceph.com/issues/63192

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

File: src/common/options/mgr.yaml.in [hunk 1]
```diff
@@ -292,6 +292,14 @@ options:
   default: true
   services:
   - mgr
+- name: mon_warn_on_pool_no_app_interval
```
Comment (idryomov) [inline src/common/options/mgr.yaml.in:295]: Just a suggestion, but `mon_warn_on_pool_no_app_grace` might be a better name -- by analogy with e.g. `mon_down_mkfs_grace` option.

File: src/mon/PGMap.cc [hunk 2]
```diff
@@ -3348,9 +3348,13 @@ void PGMap::get_health_checks(
       // application metadata is not encoded until luminous is minimum
       // required release
       if (pool.application_metadata.empty() && !pool.is_tier()) {
-        stringstream ss;
-        ss << "application not enabled on pool '" << pool_name << "'";
-        detail.push_back(ss.str());
+        const utime_t now = ceph_clock_now();
```
Comment (idryomov) [inline src/mon/PGMap.cc:3351]: Might want to make the use of `CLOCK_REALTIME` explicit with `ceph::real_clock::now()`.

**Comment by guojidan** [comment]
jenkins test make check

File: src/mon/PGMap.cc [hunk 3]
```diff
@@ -3348,9 +3348,14 @@ void PGMap::get_health_checks(
       // application metadata is not encoded until luminous is minimum
       // required release
       if (pool.application_metadata.empty() && !pool.is_tier()) {
-        stringstream ss;
-        ss << "application not enabled on pool '" << pool_name << "'";
-        detail.push_back(ss.str());
+        const auto now = ceph::real_clock::to_double(ceph::real_clock::now());
+        auto create_time = pool.get_create_time();
+        if ((now - create_time) >
```
Comment (idryomov) [inline src/mon/PGMap.cc:3353]: I might have confused you with my "might want to make the use of `CLOCK_REALTIME` explicit" comment.  Involving double isn't necessary and `const` is pretty bogus here:
```suggestion
        utime_t now(ceph::real_clock::now());
        if ((now - pool.get_create_time()) >
```

**Review by idryomov — COMMENTED** [review]
Apologies for letting this PR go stale -- this isn't really my area.  If you end up repushing, please capitalize the Fixes tag in the commit message:
```
Fixes: https://tracker.ceph.com/issues/63192
```

**Comment by idryomov** [comment]
@rzarzynski @ljflores This needs a review from Core.

**Comment by rzarzynski** [comment]
@pdvian, would you mind taking a look?

File: src/mon/PGMap.cc (hunk 3 above)
Comment (guojidan) [reply src/mon/PGMap.cc:3353]: It's okay. I am going to fix this.  😃

**Review by idryomov — APPROVED** [review]

**Comment by idryomov** [comment]
Marking for QA while this is waiting for ack from @ceph/core.

**Comment by ronen-fr** [comment]
Approved (core) (https://tracker.ceph.com/projects/rados/wiki/MAIN#httpstrackercephcomissues67878)

**Pull request merged**